modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultCodegen.java

```diff
@@ -3309,7 +3309,8 @@ public void setParameterBooleanFlagWithCodegenProperty(CodegenParameter paramete
             parameter.isPrimitiveType = true;
         } else if (Boolean.TRUE.equals(property.isFile)) {
             parameter.isFile = true;
-            parameter.isPrimitiveType = true;
+            // file is *not* a primitive type
```

Comment:
> To clarify, I want to know whether additional primitive data type is isPrimitiveType or not.

@www2k we've been treating `isFile` as a non-primitive type (which means `isPrimitiveType` set to false). It's definitely arguable as you've pointed out the following:
```
An additional primitive data type "file" is used by the Parameter Object and the Response Object to set the parameter type or the response as being a file.
```
What I noticed is that some mustache values rely on `isFile` set to true and `isPrimitiveType` set to false. My take is to keep the old logic and to revise it with a valid use case in `2.3.0` branch.

Does your enhancement rely on both `isFile` and `isPrimitiveType` set to true?